pypdf/_reader.py

```diff
@@ -651,25 +674,56 @@ def _write_field(self, fileobj: Any, field: Any, field_attributes: Any) -> None:
                 # Field attribute is N/A or unknown, so don't write anything
                 pass
 
-    def get_form_text_fields(self) -> Dict[str, Any]:
+    def get_form_text_fields(self, full_qualified_name: bool = False) -> Dict[str, Any]:
         """
         Retrieve form fields from the document with textual data.
 
         The key is the name of the form field, the value is the content of the
         field.
 
         If the document contains multiple form fields with the same name, the
-        second and following will get the suffix _2, _3, ...
+        second and following will get the suffix .2, .3, ...
+
+        full_qualified_name should be used to get full name
         """
+
+        def indexed_key(k: str, fields: dict) -> str:
+            if k not in fields:
+                return k
+            else:
+                return (
+                    k
+                    + "."
+                    + str(sum([1 for kk in fields if kk.startswith(k + ".")]) + 2)
+                )
+
         # Retrieve document form fields
         formfields = self.get_fields()
         if formfields is None:
             return {}
-        return {
-            formfields[field]["/T"]: formfields[field].get("/V")
+        if full_qualified_name:
+            ff = {
+                (field if full_qualified_name else formfields[field]["/T"]): formfields[
+                    field
+                ].get("/V")
+                for field in formfields
+                if formfields[field].get("/FT") == "/Tx"
+            }
+        else:
+            ff = {}
+            for v in formfields.values():
+                if v.get("/FT") == "/Tx":
+                    ff[indexed_key(cast(str, v["/T"]), ff)] = (
+                        cast(str, v["/V"]) if "/V" in v else None
+                    )
```

Comment:
I think that part could be simplified to:

```suggestion
        ff = {}
        for field, value in formfields.items():
            if value.get("/FT") == "/Tx":
                if full_qualified_name:
                    ff[field] = value.get("/V")
                else:
                    ff[indexed_key(cast(str, value["/T"]), ff)] = value.get("/V")
```